xwiki-pro-macros-ui/src/main/resources/XWiki/Macros/Team.xml

```diff
@@ -221,33 +219,51 @@ $services.localization.render('rendering.macro.team.description')
     }
   }
 
-  for (const teamDiv of document.getElementsByClassName('xwikiteam')) {
-    const a = teamDiv.querySelector('.xwikiteam-tools-link a');
-    if (a) {
-      a.onclick = toggleTools.bind(null, teamDiv);
-      a.parentNode.hidden = false;
-    }
-    const ul = teamDiv.querySelector('.xwikiteam-ul');
-    const tools = teamDiv.querySelector('.xwikiteam-tools');
-    if (!tools) continue;
-    const filterInput = tools.querySelector('.xwikiteam-filter');
-    if (filterInput) {
-      filterInput.addEventListener('keyup', filter.bind(filterInput, teamDiv));
-      if (filterInput.value) {
-        filter.call(filterInput, teamDiv);
-        toggleTools(teamDiv)
+  function initTools() {
+    const teamDivs = document.getElementsByClassName('xwikiteam');
+    for (let i = 0, len = teamDivs.length; i &lt; len; i++) {
+      const teamDiv = teamDivs[i];
+      const a = teamDiv.querySelector('.xwikiteam-tools-link a');
+      if (a) {
+        if (a.onclick) {
+          continue; // don't initialize already initialized team macros
+        }
+        a.onclick = toggleTools.bind(null, teamDiv);
+        a.parentNode.hidden = false;
       }
+      const ul = teamDiv.querySelector('.xwikiteam-ul');
+      const tools = teamDiv.querySelector('.xwikiteam-tools');
+      if (!tools) continue;
+      const filterInput = tools.querySelector('.xwikiteam-filter');
+      if (filterInput) {
+        filterInput.addEventListener('keyup', filter.bind(filterInput, teamDiv));
+        if (filterInput.value) {
+          filter.call(filterInput, teamDiv);
+          toggleTools(teamDiv)
+        }
+      }
+
+      bindShowUsernameCheckbox(teamDiv, teamDiv.querySelector(".xwikiteam-show-username-checkbox"));
     }
+  }
+
+  function init() {
+    initTools();
+    document.observe('xwiki:dom:updated', initTools);
```

Comment:
Don't use Prototype.js . Use jQuery for listening to custom XWiki events (as per https://www.xwiki.org/xwiki/bin/view/Documentation/DevGuide/FrontendResources/JavaScriptAPI/#HBridgingcustomXWikieventsbetweenPrototypeandjQuery ) Also, when ``xwiki:dom:updated`` is triggered you should re-initialize **only** the ``xwikiteam`` elements that have been updated.

Reply:
fixed